From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 07 Jul 2014 01:09:53 +0100
Subject: irq: Avoid ABI change in 3.14.10
Forwarded: not-needed

Move the new fields to the end of struct irq_desc and hide them from
genksyms.  Although struct irq_desc is normally allocated as part of a
static array, access is abstracted through the irq_to_desc() function
so modules don't depend on the size of the structure.

--- a/include/linux/irqdesc.h
+++ b/include/linux/irqdesc.h
@@ -54,8 +54,6 @@ struct irq_desc {
 	unsigned int		irq_count;	/* For detecting broken IRQs */
 	unsigned long		last_unhandled;	/* Aging timer for unhandled count */
 	unsigned int		irqs_unhandled;
-	atomic_t		threads_handled;
-	int			threads_handled_last;
 	raw_spinlock_t		lock;
 	struct cpumask		*percpu_enabled;
 #ifdef CONFIG_SMP
@@ -74,6 +72,10 @@ struct irq_desc {
 	int			parent_irq;
 	struct module		*owner;
 	const char		*name;
+#ifndef __GENKSYMS__
+	atomic_t		threads_handled;
+	int			threads_handled_last;
+#endif
 } ____cacheline_internodealigned_in_smp;
 
 #ifndef CONFIG_SPARSE_IRQ
